packages/@lwc/engine-core/src/framework/check-version-mismatch.ts

```diff
@@ -14,7 +14,8 @@ import { LightningElementConstructor } from './base-lightning-element';
 
 let warned = false;
 
-if (process.env.NODE_ENV === 'development') {
+// @ts-ignore
+if (process.env.NODE_ENV !== 'production' && typeof __karma__ !== 'undefined') {
```

Comment:
This is only used here:

https://github.com/salesforce/lwc/blob/90b213b6c65834fe7dd33cf1f007a2314954a7f4/packages/integration-karma/test/rendering/version-mismatch/index.spec.js#L43-L45